packages/api/src/command/medical/patient/__tests__/validate.test.ts

```diff
@@ -37,4 +38,24 @@ describe("validate", () => {
       expect(resp).toBeTruthy();
     });
   });
+
+  describe("dob", () => {
+    it("throws an error if dob is in the future", async () => {
+      const futureDate = dayjs().add(7, 'day').format('YYYY-MM-DD');
+      const patient = makePatientCreate({ dob: futureDate });
+      expect(() => validate(patient)).toThrow(`Date must be in the past: ${futureDate}`);
+    });
+
+    it("returns true when dob is valid", async () => {
+      const currentDate = dayjs().format('YYYY-MM-DD');
+      let patient = makePatientCreate({ dob: currentDate });
+      let resp = validate(patient);
+      expect(resp).toBeTruthy();
+
+      const pastDate = dayjs().subtract(7, 'day').format('YYYY-MM-DD');
+      patient = makePatientCreate({ dob: pastDate });
+      resp = validate(patient);
+      expect(resp).toBeTruthy();
+    })
```

Comment:
Can we split this into 2 tests, one to validate it works to set DOB as current date (I'd subtract 1 second to make sure we're sending a date in the "past"), and one to validate it accepts a DOB a few days in the past, please?

It's good to have individual unit tests for each scenario.

While there, it would be good to have one for current date but a minute "in the future", to validate it errors.

Reply:
@jamestran201 friendly bump on this :)

Reply:
Sorry for the late response

> While there, it would be good to have one for current date but a minute "in the future", to validate it errors.

Hmm, I wasn't able to come up with a way to do this, I haven't used TS a lot 😅 . Because there are [multiple calls to `daysjs()`](https://github.com/metriport/metriport/pull/2725/files#diff-86f92a29b7de58ea90e5ffb6f48d2cf3e42f21ad1af9a1b813ee953e74c2e6d4R37), and I think we need to make it return different stub results. The stubbed results would be one `Dayjs` object for `2024-09-21 00:00:00` and one for `2024-09-20: 23:59:00`. Do you have any tips on how this could be done?

Reply:
Hey @jamestran201 !

It would be something like this:

```ts
    it("throws an error when dob is a minute in the future", async () => {
      const futureDate = dayjs().add(1, "minute").toISOString(); // this returns a datetime representation
      const patient = makePatientCreate({ dob: futureDate });
      expect(() => validate(patient)).toThrow(`Date must be in the past: ${futureDate}`);
    })
```